src/lambda/wizards/samInitWizard.ts

```diff
@@ -274,12 +274,7 @@ export class CreateNewSamAppWizard extends Wizard<CreateNewSamAppWizardForm> {
 
         this.form.name.bindPrompter(state =>
             createNamePrompter(
-                fsutil.getNonexistentFilename(
-                    state.location!.fsPath,
-                    `lambda-${state.runtimeAndPackage!.runtime}`,
-                    '',
-                    99
-                )
+                `${`lambda-${state.runtimeAndPackage!.runtime}`}-${crypto.randomBytes(4).toString('hex')}`
```

Comment:
Why was this change made? New code is a minor UX regression.

Reply:
This was due to the code needing to be synchronous, I created a duplicate of the async method. See `getNonexistentFilename()`